File: frontend/src/lib/types/admin-categories.ts
// Mirrors backend/app/schemas/admin_categories.py exactly — keep these two in sync.

export interface CategoryCreate {
  name: string;
  description?: string | null;
  color_tag?: string;
}

export interface CategoryUpdate {
  name?: string;
  description?: string | null;
  color_tag?: string | null;
}

export interface CategoryPublic {
  id: string;
  name: string;
  description: string | null;
  color_tag: string;
  dataset_count: number;
}

File: backend/app/schemas/admin_categories.py
import uuid

from pydantic import BaseModel, Field


class CategoryCreate(BaseModel):
    name: str = Field(min_length=1, max_length=100)
    description: str | None = None
    color_tag: str = Field(default="cat-default", max_length=50)


class CategoryUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=100)
    description: str | None = None
    color_tag: str | None = Field(default=None, max_length=50)


class CategoryPublic(BaseModel):
    id: uuid.UUID
    name: str
    description: str | None
    color_tag: str
    dataset_count: int

    model_config = {"from_attributes": True}
